osbrain/proxy.py

```diff
@@ -92,7 +92,9 @@ def _ready_or_timeout(self, time0, timeout):
             exception.
         """
         try:
-            self.ping()
+            safe_status = self._safe
+            self.unsafe.ping()
+            self._safe = safe_status
```

Comment:
I don't think you need to save the `._safe` status here (if you do, then it would be a bug). Can you confirm?